### PR TITLE
Add eNear to the omni bridge

### PR DESCRIPTION
**Pull request by kiseln**
Added a standard to map custom Ethereum tokens to the Near counterparts. This is useful to migrate existing (legacy) rainbow bridge assets to the OmniBridge. The first implementation is for `eNear`.

Created a proxy (or adapter?) for eNear that will be the only address that can mint/burn eNear tokens.
- For burning. We have to transfer the tokens to the proxy address and then call eNear `transferToNear` to burn.
- For minting. We set a fake prover that always returns true. We submit a fake proof with appropriate amount and recipient to eNear's `finaliseNearToEthTransfer`.

- eNear implementation moved from: https://github.com/Near-One/near-erc20-connector/tree/main/eNear
- Add basic tests.

### CODE REVIEW

File: evm/src/eNear/contracts/ENearProxy.sol [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
...9c0ade5130111274bbd148fca204baa857aad66db092c89fdd2254d1565250d0fa1307af0bd01639b386ef85d52165cd91c0b0d7ff3b72545e6ef652030c9271ab6807b6bcd3b01d13445f2ea94746501e2044ea06f22111fe718788615318b50439789575cf3f601198df8d64579dbe8ccedf43a29b93af50ebf4af0d808cda6b8dee39764f3b03301c39900c6d93efe1755b283cd01a5f4717f6bb6c989617a0fb990c3a2b900796f0083b7cad2577c5a1d3be71aa6f56f75544327569119dc7d0df0ab7e3299b6070700fb2726881c7fa27650003873eb777a1dc21a7c0e63e227da32e64e5160a0425b0062f5c0b2a58d6cae9e53770c93a2eed9685d1c5892a7e1d05b383b68e781d90601120e09be45535aff6e794e78a45ade30f541c23369225e13b441eb58016e0e3301945a5d758ccd4d686c486c210d2ab4e5f4fa15e208d118298f3de87a1e361b4b0160f2e83a0bb4d2cd1cba6ba50c8d66208a90c08bd694e9b2c33f229e9e9a9a03000dd2a5f4062d9a92130cede984a03f81121667024cc4967d002bd889587fc6de0054ce0c34d981837f3fa9f1d99eb5905be18876bba23f6e1016603c505099403e005c0fb550cec2d64f12b537f2c2a00aa339412b39a7c6b7e9506acdcee19b43af00ea6693e8d743a36fa44009abd3261f1b484e339c01383fe684cc054b3f04384000e3829b656ca60088ebdd88090d46f1d39a2f6c0744512c5a38614c86e4bf586200273a218152b554401ec79fab2e5606126e01f3c560f2040abe4487c3bba8180100124bb839a5b9c44856054c6d80bd5229ac2b40fa8a0cd44911629f295b11f08200"
```
Comment (olga24912) [inline evm/src/eNear/contracts/ENearProxy.sol:31]: I’d prefer to move it to a constant and add a comment about the origin of this string.

File: evm/src/eNear/scripts.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...tateMutability":"view","type":"function"},{"inputs":[],"name":"totalSupply","outputs":[{"internalType":"uint256","name":"","type":"uint256"}],"stateMutability":"view","type":"function"},{"inputs":[{"internalType":"address","name":"recipient","type":"address"},{"internalType":"uint256","name":"amount","type":"uint256"}],"name":"transfer","outputs":[{"internalType":"bool","name":"","type":"bool"}],"stateMutability":"nonpayable","type":"function"},{"inputs":[{"internalType":"address","name":"sender","type":"address"},{"internalType":"address","name":"recipient","type":"address"},{"internalType":"uint256","name":"amount","type":"uint256"}],"name":"transferFrom","outputs":[{"internalType":"bool","name":"","type":"bool"}],"stateMutability":"nonpayable","type":"function"},{"inputs":[{"internalType":"uint256","name":"_amount","type":"uint256"},{"internalType":"string","name":"_nearReceiverAccountId","type":"string"}],"name":"transferToNear","outputs":[],"stateMutability":"nonpayable","type":"function"},{"inputs":[{"internalType":"bytes32","name":"","type":"bytes32"}],"name":"usedProofs","outputs":[{"internalType":"bool","name":"","type":"bool"}],"stateMutability":"view","type":"function"}]
```
Comment (kiseln) [inline evm/src/eNear/scripts.js:1]: Can we add a solidity interface instead and use it with getContractFactory?

Comment (olga24912) [reply evm/src/eNear/scripts.js:1]: I think we can. I just made it exactly as it is deployed. But maybe for the ABI, it’s not super important.

Comment (olga24912) [reply evm/src/eNear/scripts.js]: fixed: https://github.com/Near-One/omni-bridge/pull/77/commits/80e76351d7806633761c893e0fdeea2557b3bfc4

File: evm/src/eNear/contracts/ENearProxy.sol [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+import {UUPSUpgradeable} from '@openzeppelin/contracts-upgradeable/proxy/utils/UUPSUpgradeable.sol';
+import {IENear} from './IENear.sol';
+import {ICustomMinter} from '../../common/ICustomMinter.sol';
+
+contract ENearProxy is UUPSUpgradeable, AccessControlUpgradeable, ICustomMinter {
+    IENear public eNear;
+
+    bytes32 public constant MINTER_ROLE = keccak256("MINTER_ROLE");
+    bytes public nearConnector;
+    uint256 public current_receipt_id;
```
Comment (karim-en) [inline evm/src/eNear/contracts/ENearProxy.sol:15]: Please use camel case

**Review by karim-en — CHANGES_REQUESTED** [review]

File: evm/src/eNear/contracts/ENearProxy.sol (hunk 3 above)
Comment (olga24912) [reply evm/src/eNear/contracts/ENearProxy.sol]: fixed: https://github.com/Near-One/omni-bridge/pull/77/commits/72878f0006ff0055eae3c6e3f4398a41cc19b15d

**Review by karim-en — APPROVED** [review]

**Pull request merged**